### PR TITLE
Update library.yaml

### PR DESCRIPTION
**Pull request by chancenlaw**
Added:
GSE167998: flowSorted.BloodExtended.EPIC
GSE180683: CD4T and CD8T naïve and memory validation dataset
GSE180970: Umbilical cord blood artificial mixtures validation dataset

### CODE REVIEW

File: biolearn/data/library.yaml [hunk 1]
```diff
@@ -864,4 +864,283 @@ items:
...
+      memory cells, natural killers, and T regulatory cells). Application of the
+      IDOL algorithm for identifying optimal libraries for the
+      deconvolution of blood-derived DNA methylation data led to to a library
+      consisting of 1200 CpGs. The accuracy of deconvolution estimates obtained
+      using our IDOL-optimized library was evaluated using artificial mixtures
+      with known cellular composition and in samples were both whole-blood DNA
+      methylation data and FACS information were available. We further
+      showcase potential applications of our expanded library using
+      publicly available cancer, aging, and autoimmune disease data sets.
+    format: Unknown
```
Comment (sarudak) [inline biolearn/data/library.yaml:886]: It looks like the script failed to determine the format automatically. It's probably IlluminaEPIC but we should verify. Same with other unknown formats. You can check the platform on the GEO site to confirm https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167998

**Review by sarudak — COMMENTED** [review]
This looks good. Can you verify that you were able to load all these new data sources and what sanity checks you did that the output was good?

File: biolearn/data/library.yaml (hunk 1 above)
Comment (chancenlaw) [reply biolearn/data/library.yaml:886]: Yes these three datasets have the IlluminaMethylationEPIC format.

Comment (chancenlaw) [reply biolearn/data/library.yaml:886]: Sanity check on zip file [Test.zip](https://github.com/user-attachments/files/16566050/Test.zip)
Please let me know if you are able to review the tests or if you want me to send the sanity checks in another format.

**Review by sarudak — APPROVED** [review]
LGTM

**Pull request merged**